lib/rmt/file_validator.rb

```diff
@@ -0,0 +1,59 @@
+require 'fileutils'
+require 'rmt/checksum_verifier'
+
+class RMT::FileValidator
+  class << self
+    def validate_local_file(file_reference, deep_verify:)
+      verify_file_on_disk(file_reference, deep_verify).tap do |valid_on_disk|
```

Comment:
The method returns the value of `valid_on_disk`, not of `update_file_status_on_database`; the database is a necessary step but it doesn't affect the method outcome. Using `.tap` was just a way to group both steps: updating the databse only happens if a file is verified on disk, but sure it's not the only one, I can do it like:

```ruby
valid_on_disk = verify_file_on_disk(file_reference, deep_verify)
update_file_status_on_database(file_reference, valid_on_disk)
valid_on_disk
```

Reply:
@lcaparroz, I would agree with @ikapelyukhin here. `tap` may be overkill for this use-case and, while grammatically correct, falls outside its normal style and can be confusing. This may be the time to be overly verbose and less clever :(.

Reply:
Also, @lcaparroz, I can only find one usage of `deep_verify` and it's `false`. Maybe it's unneeded.

Reply:
I removed the `.tap` method calls where it is not clear the benefits of using it. About `deep_verify`, I'll evaluate if it's worth implementing the `--deep-verify` flag in this PR.